### ISSUE TITLE
Rule proposal: require-description

### ISSUE THREAD

**Issue by ota-meshi** [mysticatea/eslint-plugin-eslint-comments#40]
**Please describe what the rule should do:**

This rule reports directive comments without description.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[X] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
/* eslint-disable */
/* eslint-disable rule-name */
/* global _ */
/* eslint-env node */
```

No warning is given if the directive comment contains an description.

```js
/* eslint-disable -- description */
/* eslint-disable rule-name -- description */
/* global _ -- description */
/* eslint-env node -- description */
```

May need an option to set whether or not to warn depending on the type of directives.

**Comment by mysticatea** [comment]
Sounds good to me!

**Comment by nemoDreamer** [comment]
I was _just_ considering opening a pull-request for this! ❤️ 
Thanks, @ota-meshi !!

**Comment by dimitropoulos** [comment]
thanks again for doing this: it inspired me to do https://github.com/typescript-eslint/typescript-eslint/pull/2099
